code/modules/explorer_drone/loot.dm

```diff
@@ -0,0 +1,187 @@
+	var/mob/living/simple_animal/pet/pet = new chosen_pet_type()
+	carrier.add_occupant(pet)
+	return carrier
+
+/obj/item/antique
+	name = "antique"
+	desc = "Valuable and completly incomprehensible."
+	icon = 'icons/obj/exploration.dmi'
+	icon_state = "antique"
+
+/// Supply pack unlocker chip
+/obj/item/trade_chip
+	name = "trade contract chip"
+	desc = "Uses the station's cargo network to contact a black market supplier, allowing the purchase of a new crate type at cargo console."
+	icon = 'icons/obj/exploration.dmi'
+	icon_state = "trade_chip"
+	/// Supply pack type enabled by this chip
+	var/unlocked_pack_type
+
+/obj/item/trade_chip/Initialize(mapload, pack_type)
+	. = ..()
+	if(pack_type)
+		unlocked_pack_type = pack_type
+		var/datum/supply_pack/P = pack_type
+		name += "- [initial(P.name)]"
+
+/obj/item/trade_chip/proc/try_to_unlock_contract(mob/user)
+	var/datum/supply_pack/P = SSshuttle.supply_packs[unlocked_pack_type]
+	if(!unlocked_pack_type || !P || !P.special)
+		to_chat(user,"<span class='danger'>This chip is invalid!</span>")
+		return
+	P.special_enabled = TRUE
+	to_chat(user,"<span class='notice'>Contract accepted into nanotrasen supply database.</span>")
+	qdel(src)
+
+
+/// Two handed fire lance. Melts wall after short windup.
+/obj/item/firelance
+	name = "fire lance"
+	desc = "Melts everything in front of you. Takes a while to start and operate."
+	icon = 'icons/obj/exploration.dmi'
+	icon_state = "firelance"
+	inhand_icon_state = "firelance"
+	righthand_file = 'icons/mob/inhands/misc/firelance_righthand.dmi'
+	lefthand_file = 'icons/mob/inhands/misc/firelance_lefthand.dmi'
+	var/windup_time = 10 SECONDS
+	var/melt_range = 3
+	var/charge_per_use = 200
+	var/obj/item/stock_parts/cell/cell
+
+/obj/item/firelance/Initialize()
+	. = ..()
+	cell = new /obj/item/stock_parts/cell(src)
+	AddComponent(/datum/component/two_handed)
+
+/obj/item/firelance/attack(mob/living/M, mob/living/user, params)
+	if(!user.combat_mode)
+		return
+	. = ..()
+
+/obj/item/firelance/get_cell()
+	return cell
+
+/obj/item/firelance/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!HAS_TRAIT(src,TRAIT_WIELDED))
+		to_chat(user,"<span class='notice'>You need to wield [src] in two hands before you can fire it.</span>")
+		return
+	if(LAZYACCESS(user.do_afters, "firelance"))
+		return
+	if(!cell.use(charge_per_use))
+		to_chat(user,"<span class='warning'>[src] battery ran dry!</span>")
+	ADD_TRAIT(user,TRAIT_IMMOBILIZED,src)
+	to_chat(user,"<span class='notice'>You begin to charge [src]</span>")
+	inhand_icon_state = "firelance_charging"
+	user.update_inv_hands()
+	if(do_after(user,windup_time,interaction_key="firelance",extra_checks = CALLBACK(src, .proc/windup_checks)))
+		var/turf/start_turf = get_turf(user)
+		var/turf/last_turf = get_ranged_target_turf(start_turf,user.dir,melt_range)
+		start_turf.Beam(last_turf,icon_state="solar_beam",time=1 SECONDS)
+		for(var/turf/T in getline(start_turf,last_turf))
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names